src/Api/BaseApiController.php

```diff
@@ -60,18 +60,24 @@ protected function json($data, IDataTransformer $transformer = null): Response
     }
 
     /**
-     * Validates request and throws exception, if input data in request doesn't match expected rules
+     * Validates request and throws exception, if input data in request doesn't match expected rules.
      *
      * @param Request $request Dingo/Api request (unlike in generic Laravel Controller)
      * @param array $rules Laravel-style rules for validation https://laravel.com/docs/validation
      * @param array $messages Messages to use for specified fields, if they fail validation
      * @param array $customAttributes How to name validated attributes in messages, visible to user
      *
-     * @throws ValidationException - when input data in request does not match expected rules
+     * @throws ValidationException when input data in request does not match expected rules
+     *
+     * @return void
      */
-    public function validate(Request $request, array $rules, array $messages = [], array $customAttributes = [])
+    public function validate(Request $request, array $rules, array $messages = [], array $customAttributes = []): void
```

Comment:
Is we really need this method? As i know we use FormRequests rules to validate format of passed data and EntityService validates business-rules. IMHO no need to add one more validation here. Backward compatibility already broken

Reply:
Good suggestion :+1: